### PR TITLE
update output types of publishing assets, add cloud sensors

### PR DESCRIPTION
**Pull request by yongrenjie**
WIP

Still to do:

- [x] metrics asset
- [x] metrics IO manager 🫠
  - okay this was surprisingly easy
- [x] the cloud sensors run every X seconds even for assets which haven't been re-materialised since the last run. Need to figure out how to stop that.
- [ ] add validation inside the IO managers' handle_output section to make sure that the inputs have the right types, satisfy certain invariants, etc.

### CODE REVIEW

File: python/popgetter/assets/be/census_geometry.py [hunk 1]
```diff
@@ -113,55 +109,87 @@ def geometry(
...
     )
 
-    return geometry_metadata_df, region_geometries, region_names
+    return geometries_to_return
+
+
+@asset(key_prefix=asset_prefix)
+def source_data_release_munip(
+    geometry: list[tuple[GeometryMetadata, gpd.GeoDataFrame, pd.DataFrame]]
+) -> SourceDataRelease:
```
Comment (sgreenbury) [inline python/popgetter/assets/be/census_geometry.py]: Would it work returning the `SourceDataRelease`s as a list here with the IO manager converting this list of metadata to a `pd.DataFrame` and then to parquet?

File: python/popgetter/assets/be/census_derived.py [hunk 2]
```diff
@@ -174,142 +186,158 @@ def filter_needed_catalog(
...
 
-
-@multi_asset(
+@asset(
     partitions_def=dataset_node_partition,
     ins={
-        "source_table": AssetIn(
-            key_prefix=asset_prefix, partition_mapping=needed_dataset_mapping
+        "source_metrics_by_partition": AssetIn(
+            key_prefix=asset_prefix, partition_mapping=IdentityPartitionMapping()
```
Comment (sgreenbury) [inline python/popgetter/assets/be/census_derived.py:246]: I think beyond the scope of this PR but looking at this again, wondering if more generally/other countries we need to specify the needed partitions and additional partition mapping.

File: python/popgetter/assets/be/census_geometry.py (hunk 1 above)
Comment (yongrenjie) [reply python/popgetter/assets/be/census_geometry.py]: Now takes class, list of class or dict of class

**Comment by sgreenbury** [comment]
Noting on areas to be covered in other issues:
- Currently `test_cloud_outputs.py` is commented out/skipped since the implementation has changed. This will be revised in new issue TBD
- Update SAS token to use `EnvVar` (see dagster [docs](https://docs.dagster.io/guides/dagster/using-environment-variables-and-secrets))
- Refactoring the Belgium DAG with config expressed not in in an asset for needed datasets and the transformations for derived metrics
- See [above](https://github.com/Urban-Analytics-Technology-Platform/popgetter/pull/92#issue-2299995447) re. @yongrenjie comment on validation for IO managers input

File: python/popgetter/assets/be/census_derived.py (hunk 2 above)
Comment (sgreenbury) [reply python/popgetter/assets/be/census_derived.py:246]: See new issue #95

**Pull request merged**